docs/docsite/rst/playbooks_loops.rst

```diff
@@ -676,6 +676,8 @@ This differs from the data structure returned when using ``register`` without a
         ]
     }
 
+.. note:: The data structure returned when running in ``--check`` mode will look somewhat different.
```

Comment:
it not always WILL be, but it CAN be, depends on the module. specifically _facts modules should return the same data, but modules that affect change and support check mode, might differ as they didn't really carry out their actions.